marketplacetests/marketplace/regions/add_review.py

```diff
@@ -3,22 +3,24 @@
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
 from marionette.by import By
-from gaiatest.apps.base import Base
 
+from marketplacetests.marketplace.app import Marketplace
+from marketplacetests.marketplace.regions.app_details import Details
 
-class AddReview(Base):
+class AddReview(Marketplace):
     """
     Page for adding reviews.
     """
 
+    _page_loaded_locator = (By.CSS_SELECTOR, 'form.add-review-form')
+
     _add_review_input_field_locator = (By.ID, 'review-body')
     _submit_review_button_locator = (By.CSS_SELECTOR, 'button[type="submit"]')
-    _review_box_locator = (By.CSS_SELECTOR, '.add-review-form')
     _rating_locator = (By.CSS_SELECTOR, ".ratingwidget.stars-0 > label[data-stars='%s']")
 
     def __init__(self, marionette):
-        Base.__init__(self, marionette)
-        self.wait_for_element_displayed(*self._review_box_locator)
+        Marketplace.__init__(self, marionette)
+        self.wait_for_page_loaded()
```

Comment:
I wanted to do that, but the problem is that it won't work for the base `Marketplace` object, as it is not on a page when it is initialized. I thought about putting in a conditional to check whether it is the base object or not during `__init__` but that seemed too hacky. I could create another, intermediate object between `Marketplace` and the various `region` objects to serve this purpose, but I feared you wouldn't like that either. I totally agree that getting rid of this duplication would be good. What do you suggest?